CONTRIBUTING.md

```diff
@@ -14,13 +14,53 @@ We use pull requests for code contributions. To submit a pull request:
 
 1. Fork the repository.
 2. Create a new branch for your feature or bug fix.
-3. Make your changes and commit them with clear and concise messages.
+3. Make your changes and commit them with clear and concise messages. Refer [Making Commits](#making-commits)
 4. Push your changes to your fork.
 5. Submit a pull request to the main repository, detailing the changes you've made and referencing any relevant issues.
 
 ## Code Style
 
-Please follow the existing code style and conventions used in the project. If you're unsure, feel free to ask for clarification in the pull request or issue comments.
+Please follow the existing code style and conventions used in the project. If you're unsure, feel free to ask for clarification in the pull request or issue comments. For linting and all refer [Making Commits](#making-commits)
+
+## Making Commits
+
+1. Pre-commit should be installed as a dev dependency already. If not then Create a virual environment [venv](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#create-and-use-virtual-environments) or [pyenv](https://github.com/pyenv/pyenv?tab=readme-ov-file#installation) and run the follwoing command from project root dir:
+
+    ```
+    pip install -r backend/dev-requirements.txt --upgrade
+    ```
+    Also install the eslint for both `cli/` and `webserver/`. Please use node 16 for that
+    *NOTE*: If NVM is not installed, install using [nvm](https://github.com/nvm-sh/nvm?tab=readme-ov-file#installing-and-updating)
+    ```
+    nvm install 16
+    nvm use 16
+    ```
+    ```
+    cd cli
+    yarn add eslint@^8.40.0 eslint-config-next@13.5.6 eslint-plugin-import@^2.29.0 eslint-plugin-prettier@^5.0.1 eslint-plugin-react@^7.29.4 eslint-plugin-unused-imports@^3.0.0 --dev
```

Comment:
No need to install `eslint-config-next` for cli as it doesn't depend on nextjs. And running this command makes changes in the `package.json` for `cli/`. It'd be great if you could push those changes in this PR itself.